packages/react-query/src/shared/proxy/utilsProxy.ts

```diff
@@ -158,6 +163,15 @@ type DecorateProcedure<
   getInfiniteData(
     input?: inferProcedureInput<TProcedure>,
   ): InfiniteData<inferTransformedProcedureOutput<TProcedure>> | undefined;
+
+  /**
+   * Method to extract the query key for a procedure
+   * @link https://trpc.io/docs/useContext#-the-function-i-want-isnt-here
+   */
+  getQueryKey(
+    input: inferProcedureInput<TProcedure>,
+    type: QueryType,
+  ): QueryKey;
```

Comment:
On my phone so I might be missing something, but wouldn't it make sense to have both arguments optional?

Reply:
Cc @TkDodo wdyt?

Reply:
Why would we default to? "query"?

The input would be optional if the procedure's input is optional, but we need the type (I wrote a bit about this up top or in discord with @sachinraja)

Reply:
If it's getting the key for sale of invalidation or resetting, `any` is a good default

If its for getting or changing the cache `any` might not be the best

Reply:
what is `"any"` going to return? does it match both `query` and `infinite` ? If so I think it would be a good default for `getQueryKey`

Reply:
`any` returns the query key without the type object in the key, so as long as fuzzy matching is enabled for the methods then it will match both!

Do all methods match fuzzy? Or for the "query" versions only match exact? This may need some documentation!

Reply:
> Do all methods match fuzzy?

all methods _can_, but it's not the default everywhere. Generally speaking:
- everything that has a plural in its name is fuzzy, like `getQueriesData`, `invalidateQueries` or `setQueriesDefault`
- everything that is singular defaults to exact, like `getQueryData`

Reply:
The fuzzy matching behaviour queried above for `xxxxxQuery` type methods also effects if the `input` param should be optional!... Although I think it should be as I know some deffinitely fuzzy match!

Reply:
Hmmn this may mean making things optional / default to a query key of `any` for the main path I expect this being used for (we provide many of the tanstack methods for individual queries already so this is likely to be used in the fuzzy matching ones!) but then put some good documentation on this noting the gotcha between needing exact keys for methods that are named `xxxxxQuery`